rust/oak/src/grpc/mod.rs

```diff
@@ -16,12 +16,21 @@
 
 //! Functionality to help Oak Nodes interact with gRPC.
 
+pub use crate::proto::code::Code;
 use crate::{proto, wasm, Handle, OakStatus, ReadHandle, WriteHandle};
 use protobuf::{Message, ProtobufEnum};
 
 /// Result type that uses a [`proto::status::Status`] type for error values.
 pub type Result<T> = std::result::Result<T, proto::status::Status>;
```

Comment:
Maybe also implement `Into<Result<T>>` for `proto::status::Status`? Then you can just call `.into` (or maybe it  even gets automatically converted upon return)?

Reply:
Actually better to implement `From<proto::status::Status` for `Result<T>`: https://doc.rust-lang.org/std/convert/trait.From.html

Reply:
I didn't think you could implement an external trait on an external struct?  Or am I misremembering?

Reply:
I think the rule is that at least one needs to be in the local crate, which you are right here is not the case. Also there is already this https://doc.rust-lang.org/std/result/enum.Result.html#impl-From%3CT%3E for the "ok" implicit conversion so probably would not be possible to overload it so it also works for the err case.

Reply:
(Our previous comments crossed, I think).

Hmm, I worry that `From<proto::status::Status>` has a slight foot-gun possibility because not all `Status` values are errors &ndash; what should the `Result<T>` emitted `from` an OK status input be?